tools/cli/commands/docker.js

```diff
@@ -3,25 +3,393 @@
  */
 import { spawnSync } from 'child_process';
 import chalk from 'chalk';
+import { createWriteStream, existsSync, copyFileSync, readFileSync, writeFileSync } from 'fs';
+import yaml from 'js-yaml';
+
+const dockerFolder = `tools/docker`;
 
 /**
- * Default handler for the monorepo Docker commands.
+ * Sets default options that are common for most of the commands
  *
- * @param {object} argv - Arguments passed.
+ * @param {object} yargs - Yargs
+ * @returns {object} Modified Yargs object
  */
-const defaultDockerCmdHandler = argv => {
-	const args = process.argv.slice( 4 ); // node pnpm run docker <cmd> [args..]
+const defaultOpts = yargs =>
+	yargs
+		.option( 'type', {
+			alias: 't',
+			default: 'dev',
+			describe: 'Container type',
+		} )
+		.option( 'name', {
+			alias: 'n',
+			describe: 'Project name',
+		} )
+		.option( 'port', {
+			alias: 'p',
+			describe: 'WP port',
+		} )
+		.option( 'ngrok', {
+			type: 'bool',
+			describe: 'Flag to launch ngrok process',
+		} );
+
+/**
+ * Gets a project name from the passed arguments. Defaults to 'dev' if not specified.
+ *
+ * @param {object} argv - Yargs
+ * @returns {string} Project name
+ */
+const getProjectName = argv => {
+	let project = 'dev';
+	if ( argv.type === 'e2e' ) {
+		project = argv.name ? argv.name : 'e2e';
+	}
+
+	return project;
+};
+
+/**
+ * Builds a map of ENV variables for specified configuration
+ *
+ * @param {object} argv - Yargs
+ * @returns {object} key-value pairs of ENV variables
+ */
+const buildEnv = argv => {
+	const envOpts = {};
+	if ( argv.type === 'e2e' ) {
+		envOpts.PORT_WORDPRESS = argv.port ? argv.port : 8889;
+	}
+
+	envOpts.COMPOSE_PROJECT_NAME = getProjectName( argv );
```

Comment:
Would it make sense that if `dev` is the `getProjectName` value, we set this to `jetpack` to maintain BC with existing volumes?

Related, could we also prefix it with `jetpack_` generally, e.g. `jetpack_wordpress` for `dev` and `jetpack_XXX_wordpress` for `e2e` and any others?

Reply:
Actually it isn't quite simple since `docker` was our old name which wasn't really helpful. Let me push up a quick change.

Reply:
yeah, I'm not sure it would be easy to maintain backward compatibility :/